encrypted-media/util/drm-messagehandler.js

```diff
@@ -0,0 +1,230 @@
+
+    this.messagehandler = MessageHandler.prototype.messagehandler.bind( this );
+    if ( this._drmconfig && this._drmconfig.certificate ) {
+        this.servercertificate = stringToUint8Array( atob( this._drmconfig.certificate ) );
+    }
+}
+
+MessageHandler.prototype.messagehandler = function messagehandler( messageType, message) {
+
+    const keySystems = {
+        'com.widevine.alpha': {
+            responseType: 'json',
+            getLicenseMessage: function(response) {
+                return base64DecodeToUnit8Array( response.license );
+            },
+            getErrorResponse: function(response) {
+                return response;
+            },
+            getLicenseRequestFromMessage: function(message) {
+                return new Uint8Array(message);
+            },
+            getRequestHeadersFromMessage: function(/*message*/) {
+                return null;
+            }
+        },
+        'com.microsoft.playready': {
+            responseType: 'arraybuffer',
+            getLicenseMessage: function(response) {
+                return response;
+            },
+            getErrorResponse: function(response) {
+                return String.fromCharCode.apply(null, new Uint16Array(response));
+            },
+            getLicenseRequestFromMessage: function(message) {
+                var msg,
+                    xmlDoc;
+                var licenseRequest = null;
+                var parser = new DOMParser();
+                var dataview = new Uint16Array(message);
+
+                msg = String.fromCharCode.apply(null, dataview);
+                xmlDoc = parser.parseFromString(msg, 'application/xml');
+
+                if (xmlDoc.getElementsByTagName('Challenge')[0]) {
+                    var Challenge = xmlDoc.getElementsByTagName('Challenge')[0].childNodes[0].nodeValue;
+                    if (Challenge) {
+                        licenseRequest = atob(Challenge);
+                    }
+                }
+                return licenseRequest;
+            },
+            getRequestHeadersFromMessage: function(message) {
+                var msg,
+                    xmlDoc;
+                var headers = {};
+                var parser = new DOMParser();
+                var dataview = new Uint16Array(message);
+
+                msg = String.fromCharCode.apply(null, dataview);
+                xmlDoc = parser.parseFromString(msg, 'application/xml');
+
+                var headerNameList = xmlDoc.getElementsByTagName('name');
+                var headerValueList = xmlDoc.getElementsByTagName('value');
+                for (var i = 0; i < headerNameList.length; i++) {
+                    headers[headerNameList[i].childNodes[0].nodeValue] = headerValueList[i].childNodes[0].nodeValue;
+                }
+                // some versions of the PlayReady CDM return 'Content' instead of 'Content-Type'.
+                // this is NOT w3c conform and license servers may reject the request!
```

Comment:
I guess HTTP, but the relevant thing is not anything about the specifications but that the server we are using expects `Content-Type`, but some PlayReady libraries give the header value `Content`. If the PlayReady integrations on the browsers we are testing give `Content-Type` we could remove this code.